tools/im2rec.py

```diff
@@ -1,238 +1,265 @@
-                write_record(args, image_list, f)
-
-
-if __name__ == '__main__':
-    main()
+# -*- coding: utf-8 -*-
+# add file encoding here
+
+import os
+import sys
+
+curr_path = os.path.abspath(os.path.dirname(__file__))
+sys.path.append(os.path.join(curr_path, "../python"))
+import mxnet as mx
+import random
+import argparse
+import cv2
+import time
+
+def list_image(root, recursive, exts):
+    image_list = []
+    if recursive:
+        cat = {}
+        for path, subdirs, files in os.walk(root, followlinks=True):
+            subdirs.sort()
+            print(len(cat), path)
+            for fname in files:
+                fpath = os.path.join(path, fname)
+                suffix = os.path.splitext(fname)[1].lower()
+                if os.path.isfile(fpath) and (suffix in exts):
+                    if path not in cat:
+                        cat[path] = len(cat)
+                    image_list.append((len(image_list), os.path.relpath(fpath, root), cat[path]))
+    else:
+        for fname in os.listdir(root):
+            fpath = os.path.join(root, fname)
+            suffix = os.path.splitext(fname)[1].lower()
+            if os.path.isfile(fpath) and (suffix in exts):
+                image_list.append((len(image_list), os.path.relpath(fpath, root), 0))
+    return image_list
+
+
+def write_list(path_out, image_list):
+    with open(path_out, 'w') as fout:
+        n_images = xrange(len(image_list))
+        for i in n_images:
+            line = '%d\t' % image_list[i][0]
+            for j in image_list[i][2:]:
+                line += '%d\t' % j
+            line += '%s\n' % image_list[i][1]
+            fout.write(line)
+
+
+def make_list(args):
+    image_list = list_image(args.root, args.recursive, args.exts)
+    if args.shuffle is True:
+        random.seed(100)
+        random.shuffle(image_list)
+    N = len(image_list)
+    chunk_size = (N + args.chunks - 1) / args.chunks
+    for i in xrange(args.chunks):
+        chunk = image_list[i * chunk_size:(i + 1) * chunk_size]
+        if args.chunks > 1:
+            str_chunk = '_%d' % i
+        else:
+            str_chunk = ''
+        sep = int(chunk_size * args.train_ratio)
+        sep_test = int(chunk_size * args.test_ratio)
+        write_list(args.prefix + str_chunk + '_test.lst', chunk[:sep_test])
+        write_list(args.prefix + str_chunk + '_train.lst', chunk[sep_test:sep_test + sep])
+        write_list(args.prefix + str_chunk + '_val.lst', chunk[sep_test + sep:])
+
+
+def read_list(path_in):
+    image_list = []
+    with open(path_in) as fin:
+        for line in fin.readlines():
+            line = [i.strip() for i in line.strip().split('\t')]
+            item = [int(line[0])] + [line[-1]] + [int(i) for i in line[1:-1]]
+            image_list.append(item)
+    return image_list
+
+
+# Changed the original function write_record cause the multiprocessing must be in the __main__ process in Windows, otherwise the Pickle

```

Comment:
Changed the original function write_record cause the multiprocessing must be in the __main__ process in Windows, otherwise the Pickle.Error would happen